File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<vector<int>> queries) {
    vector<int> answer;
    
    for(int i = 0; i < queries.size(); i++) {
        int s = queries[i][0];
        int e = queries[i][1];
        int k = queries[i][2];
        int min = 1000001;
        bool find = false;
        
        for(int j = s; j <= e; j++) {
            if(k < arr[j] && arr[j] < min) {
                min = arr[j];
                find = true;
            }
        }
        
        if(find) answer.push_back(min);
        else answer.push_back(-1);
    }
    
    return answer;
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
import java.util.*;

class Solution {
    public ArrayList solution(int[] arr, int[][] queries) {
        ArrayList<Integer> list = new ArrayList<>();
        
        for(int i = 0; i < queries.length; i++) {
            int s = queries[i][0];
            int e = queries[i][1];
            int k = queries[i][2];
            
            int min = Integer.MAX_VALUE;
            for(int t = s; t <= e; t++) {
                if(arr[t] < min && arr[t] > k) {
                    min = arr[t];
                }
            }
            
            list.add(min == Integer.MAX_VALUE ? -1 : min);
        }
        
        return list;
    }
}